api/auth.go

```diff
@@ -24,10 +24,36 @@ package api
 import (
 	"context"
 	"errors"
+	"net/http"
 
 	connect "github.com/bufbuild/connect-go"
 )
 
+// http auth handler
+func (a *Api) NewAuthHandler(h http.Handler) http.Handler {
```

Comment:
Consider splitting the token instead of accesing via slice indexing:

```go
splitToken := strings.Split(reqToken, "Bearer")
if len(splitToken) != 2 {
    // Error: Bearer token not in proper format
}

// reqToken exists now, so it's safe to access
reqToken = strings.TrimSpace(splitToken[1])
// further authentication logic
```